### ISSUE TITLE
External Allowed Apps not working

### ISSUE THREAD

**Issue by ar664** [schwabe/ics-openvpn#1469]
# General information
1. Android 12 ( One UI 4 )
2. Stock OTA
3. Samsung S10
4. 0.7.33
 
# Description of the issue

Doesn't remember/store trusted applications. For the "Trust this application" ui, I have to constantly press the checkbox and ok for the Tasker. 

The following is what I see after I accept it as an allowed app. It basically shows up as an empty list of allowed apps. :

![Screenshot_20220315-135531_OpenVPN for Android](https://user-images.githubusercontent.com/11823835/158441603-d32f4fb5-3440-4f39-a242-c83fe68c32ca.jpg)
![Screenshot_20220315-135540_OpenVPN for Android](https://user-images.githubusercontent.com/11823835/158441601-3ff59530-825e-4c4a-a207-8ed30318fab5.jpg)
![Screenshot_20220315-135609_OpenVPN for Android](https://user-images.githubusercontent.com/11823835/158441597-7b4d7433-5175-42c7-9f3c-c8187180398c.jpg)

# Log (if applicable)

```
N/A
```

# Configuration file

```
N/A
```

**Comment by ar664** [comment]
Maybe this might be a possible source. Idk what api you are targeting with this app.

https://github.com/schwabe/ics-openvpn/blob/97546b9f8b51c0dc60d591b96ae835f00f2b6805/main/src/main/java/de/blinkt/openvpn/api/RemoteAction.java#L69
https://developer.android.com/reference/android/app/Activity#getCallingPackage()

**Comment by schwabe** [comment]
Since I am not familar with the Tasker setup how are you calling the app from Tasker?

The app targets API 30 or 32 (something very recent). Jelly_bean_mr2 is 4.3. YOur Android 10 is new enough not be affected by that bug.

**Comment by ar664** [comment]
I found out Tasker doesn't use the startActivityForResult() function. I used the System->Intent function in Tasker with the de.blinkt.api you have. It worked, but it wasn't perfect.

Thankfully I found a solution, which was to install a plugin that does it the intended way. It looks to be alittle outdated, so I don't know how long it will work for.
https://play.google.com/store/apps/details?id=com.ffrog8.openVpnTaskerPlugin

I would suggest adding this caveat in the documentation. Sorry about the confusion, hopefully this will be helpful for future people.
